src/rdb.c

```diff
@@ -388,18 +388,24 @@ ssize_t rdbSaveLzfBlob(rio *rdb, void *data, size_t compress_len, size_t origina
 ssize_t rdbSaveLzfStringObject(rio *rdb, unsigned char *s, size_t len) {
     size_t comprlen, outlen;
     void *out;
+    static void *buffer = NULL;
 
     /* We require at least four bytes compression for this to be worth it */
     if (len <= 4) return 0;
     outlen = len - 4;
-    if ((out = zmalloc(outlen + 1)) == NULL) return 0;
+    if (outlen < LZF_STATIC_BUFFER_SIZE) {
+        if (!buffer) buffer = zmalloc(LZF_STATIC_BUFFER_SIZE);
+        out = buffer;
+    } else {
+        if ((out = zmalloc(outlen + 1)) == NULL) return 0;
+    }
     comprlen = lzf_compress(s, len, out, outlen);
     if (comprlen == 0) {
-        zfree(out);
+        if (outlen >= LZF_STATIC_BUFFER_SIZE) zfree(out);
```

Comment:
Consider: `if (out != buffer) zfree(out);` it is more clear.


And maybe to avoid the code duplication:
```
    comprlen = lzf_compress(s, len, out, outlen);
    ssize_t nwritten = comprlen > 0 ? rdbSaveLzfBlob(rdb, out, comprlen, len) :0;
    if (out != buffer) zfree(out);
    return nwritten;
```